lightning/src/ln/channel.rs

```diff
@@ -2962,6 +2982,33 @@ impl<Signer: Sign> Channel<Signer> {
 			panic!("Cannot update fee while peer is disconnected/we're awaiting a monitor update (ChannelManager should have caught this)");
 		}
 
+		// Before proposing a feerate update, check that we can actually afford the new fee.
+		let inbound_stats = self.get_inbound_pending_htlc_stats(Some(feerate_per_kw));
+		let outbound_stats = self.get_outbound_pending_htlc_stats(Some(feerate_per_kw));
+		let keys = if let Ok(keys) = self.build_holder_transaction_keys(self.cur_holder_commitment_transaction_number) { keys } else { return None; };
+		let (_, _, num_htlcs, _, holder_balance_msat, _) = self.build_commitment_transaction(self.cur_holder_commitment_transaction_number, &keys, true, true, logger);
+		let total_fee_sat = Channel::<Signer>::commit_tx_fee_sat(feerate_per_kw, num_htlcs + CONCURRENT_INBOUND_HTLC_FEE_BUFFER as usize);
```

Comment:
Hmm, this is a bit confusing, maybe we should add a way to pass a new fee to `build_commitment_transaction` and just get the fee from there? Specifically, its not clear to me that `build_commitment_transaction` is returning a balance that is pre-fee, and it could just return a balance that is post-fee, post-dust threshold.

Reply:
I would like to do so though note the fees applied for the HTLC sending checks account for the FEE_SPIKE_BUFFER, which are different fees than the included-in-transaction one from `build_commitment_transaction`. Ultimately we could merge `next_local_commit_fee`/`next_remote_commit_fee`/`build_commitment_transaction` though it's better left as a follow-up, I think.